### PR TITLE
New style CReLU

### PR DESCRIPTION
**Pull request by hvy**
New style CReLU to support double backprop.

### CODE REVIEW

**Comment by unnonouno** [comment]
jenkins test this please

File: chainer/functions/activation/crelu.py [hunk 1]
```diff
@@ -25,22 +26,23 @@ def get_output_shape(self, input_shape):
...
-        xp = cuda.get_array_module(x)
-        gy, = gy
-        gy_former, gy_latter = xp.split(gy, 2, axis=self.axis)
-        return gy_former * (x > 0) - gy_latter * (-x > 0),
+    def backward(self, indexes, grad_outputs):
+        x, = self.get_retained_inputs()
+        gy, = grad_outputs
+        gy_former, gy_latter = chainer.functions.split_axis(
+            gy, 2, axis=self.axis)
+        return gy_former * (x.data > 0) - gy_latter * (-x.data > 0),
```
Comment (unnonouno) [inline chainer/functions/activation/crelu.py]: `(x.data < 0)` instead of `(-x.data > 0)`

Comment (hvy) [reply chainer/functions/activation/crelu.py]: Fixed!

**Comment by hvy** [comment]
Thank you for the review. I fixed the code accordingly.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
Rebased on top of master.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**